plugin/helpers_bots.go

```diff
@@ -18,11 +18,24 @@ func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error)
 	}
 
 	// If we fail for any reason, this could be a race between creation of bot and
-	// retreval from anouther EnsureBot. Just try the basic retrieve existing again.
+	// retreval from another EnsureBot. Just try the basic retrieve existing again.
 	defer func() {
 		if retBotId == "" || retErr != nil {
-			time.Sleep(time.Second)
-			botIdBytes, err := p.API.KVGet(BOT_USER_KEY)
+			var err error
+			var botIdBytes []byte
+
+			// this func is the operation that will be retried if
```

Comment:
nit: I'd propose removing the documentation of how `utils.ProgressiveRetry` works. I'm fine with it being hard-coded for now, and if future clients want to customize the retry logic, we can add a parameter which will self-document what happens.

Personally, I'd get rid of the comment altogether, and just inline the very callback as:
```go
err = utils.ProgressiveRetry(func() error {
    // ...
})
```